packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -277,6 +283,15 @@ class TreeViewExtImpl<T> implements Disposable {
         this.proxy.$setDescription(this.treeViewId, this._description);
     }
 
+    private _badge?: ViewBadge = undefined;
+    get badge(): ViewBadge | undefined {
+        return this._badge;
+    }
+    set badge(badge: ViewBadge | undefined) {
+        this._badge = badge;
+        this.proxy.$setBadge(this.treeViewId, this._badge);
```

Comment:
This is a potential problem: the `badge` object is created by the plugin. It could contain arbitrary structures, which might no be serializable via our rpc algorithm (this has happened before 🤷) We need to copy the values like
```typescript
this.proxy.$setBadge(this.treeViewId, {value: badge.value, tooltip: badge.tooltip});
```
```

Reply:
This is adressed in the new version, with a slight change due to badge being potentially `undefined`:
`this.proxy.$setBadge(this.treeViewId, badge ? { value: badge.value, tooltip: badge.tooltip } : undefined);`

There may be a shorter syntax I am not aware of.